src/main/java/com/eleks/academy/whoami/core/state/SuggestingCharacters.java

```diff
@@ -131,7 +132,10 @@ private Map<String, String> shitftCharactersBy(Map<String, String> playerToChara
 	private void fillPlayersWithShuffledCharacters(Map<String, SynchronousPlayer> players,
 												   Map<String, String> playerCharacterShuffled) {
 		for (var playerCharacters : playerCharacterShuffled.entrySet()) {
-			players.get(playerCharacters.getKey()).setCharacter(playerCharacters.getValue());
+			CharacterSuggestion suggestion = new CharacterSuggestion();
+			suggestion.setCharacter(playerCharacters.getValue());
+			suggestion.setNickName(this.players.get(playerCharacters.getKey()).getNickName());
```

Comment:
we shouldn't change nicknames only set new character to player, here as you see nickname also will be updated